Syntax for function1 kleisli-composition [help for tests appreciated]

valenterry opened the pull request:
This PR aims to add syntax for easily composing functions without wrapping them into `Kleisli` manually, based on a gitter conversation: https://gitter.im/typelevel/cats?at=5e971c478821056d3e124e4a

I looked into the tests, but I'm unsure what tests to write for the syntax methods that are added in this PR. Optimally I would like to write a property test / law in the style of `forAll{ f >=> g <-> Kleisli(f) >>> Kleisli(g) }`

TODO
- [ ] Add tests for the new syntax methods

LukaJCB commented:
We could skip the wrapping in `Kleisli` by only requiring `FlatMap[F]`

LukaJCB commented:
In terms of test, I think some simple doctests would be enough I think (along with the syntax tests you already wrote) :)

valenterry commented:
> In terms of test, I think some simple doctests would be enough I think (along with the syntax tests you already wrote) :)

Addressed that and changed it to use `FlatMap` constraint

valenterry commented on core/src/main/scala/cats/syntax/function1.scala:
```diff
@@ -30,4 +33,47 @@ trait Function1Syntax {
+     * res2: Option[Int] = None
+     * }}}
+     */
+    def >=>[C](g: B => F[C]): A => F[C] = a => FlatMap[F].flatMap(f(a))(g)
```
Should we @inline this?

barambani replied:
my two cents here, the other syntax ops are not inlined in general so I wouldn't do it.

bplommer commented:
I think this would benefit from also having non-symbolic names - maybe `andThenF` / `composeF`?

valenterry commented:
> In Cats we've been leaning towards having non-symbolic names (with symbolic alias) in most if not all cases. Can we stick to that convention here too?

Sure! Any preference for the method name? `andThenF` makes sense to me or `andThenK` (which was/is used in scalaz AFAIR)?

LukaJCB commented:
I like `composeF` and `andThenF` :)
